### PR TITLE
Added test framework and some initial tests

### PR DESCRIPTION
**Pull request by cgutteridge**
Adds vitest and some tests

### CODE REVIEW

File: test/Helpers.test.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+describe('Helpers', () => {
+  describe('ensureArray()', () => {
+    test('When passed an array: return the array unchanged', () => {
+      expect(Helpers.ensureArray([1, 2, 3])).toStrictEqual([1, 2, 3])
+    })
+    test('When passed a non array: return it wrapped in an array', () => {
+      expect(Helpers.ensureArray(242)).toStrictEqual([242])
+    })
+  })
+  describe('tableToRecords()', () => {
```
Comment (dp11g14) [inline test/Helpers.test.js:15]: Out of interest was this lack of line breaks between methods enforced by StandardJS? If not can you add a line break to make it more readable?

File: test/FieldMapper.test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+      }).toThrowError('Fields must have label, unless they are type \'ignore\'')
+    })
+
+    test('When not passed a label on a field of type "ignore": no error is thrown', () => {
+      expect(() => {
+        const fieldMapper = new FieldMapper({ id: 'Title', type: 'ignore' })
+      }).not.toThrowError()
+    })
+
+    test('When source_headings is a list including AUTO: throw an erorr', () => {
```
Comment (dp11g14) [inline test/FieldMapper.test.js]: There is a typo in erorr here. Also, maybe use the phrase 'an error is thrown' as you have in previous test names (just to be consistent).

File: test/BufferToTable.test.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import { describe, expect, test } from 'vitest'
+import { BufferToTable } from '../lib/CapeMapper/BufferToTable.js'
+import { readFileSync } from 'fs'
+
+describe('BufferToTable', () => {
+  describe('convert_xlsx()', () => {
+    test('When called with a valid XLSX byte-stream: Returns the contents as a data structure', () => {
+      const buffer = readFileSync('test/data/Test.xlsx')
```
Comment (dp11g14) [inline test/BufferToTable.test.js:8]: Has this test file been committed?

File: test/FieldMapper.test.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      expect(() => {
+        const fieldMapper = new FieldMapper({ id: 'title', label: 'Title', source_headings: ['foo', 'AUTO'] })
+      }).not.toThrowError()
+    })
+    /* eslint-enable no-unused-vars */
+
+    test('When multiple is not set: default it to false', () => {
```
Comment (dp11g14) [inline test/FieldMapper.test.js]: Maybe rename the test here to be a statement of what happens (i.e. a specification) e.g.

_When multiple is not set then multiple defaults to false_

File: test/FieldMapper.test.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    test('When a string not containing a number is processed by a integer field mapper: it is cast to NaN', () => {
+      const fieldMapper = new FieldMapper({ label: 'Year', type: 'integer' })
+      expect(fieldMapper.process_single_raw_value('foobar')).toStrictEqual(NaN)
+    })
+    test('When an integer is processed by a integer field mapper: it is returned unchanged', () => {
+      const fieldMapper = new FieldMapper({ label: 'Year', type: 'integer' })
+      expect(fieldMapper.process_single_raw_value(23)).toStrictEqual(23)
+    })
+
+    test('When an string of over 10 characters is processed by a date field mapper: it is returned truncated', () => {
```
Comment (dp11g14) [inline test/FieldMapper.test.js]: Maybe change to:

_When a string of over 10 characters is processed by a date field mapper: it returns only the first 10 characters_

**Review by dp11g14 — COMMENTED** [review]
Good work :) Just a few minor comments/suggestions.

File: test/Helpers.test.js (hunk 1 above)
Comment (cgutteridge) [reply test/Helpers.test.js:15]: it enforces no padding at the start and end of a block, but I've added lines between tests now.

**Pull request merged**